### DISCUSSION TITLE
Syntax for getting the repository_dispatch type

### DISCUSSION THREAD

**Discussion by m-hilgendorf** [community/community#26994] [Actions]
<p>Roughly what I need is this:</p>
<pre><code>on:
  repository_dispatch:
    types: [this, that, the-other-thing]
env:
  DISPATCH_TYPE: ${{ github.event.??? }}
</code></pre>
<p>But I can’t figure out the proper incantation to find out what the repository_dispatch type was that triggered my action, and I can’t find it in the documentation anywhere.</p>
<p>Any suggestions?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/m-hilgendorf">@m-hilgendorf</a> ,</p>
<p>Sure, you can use the property " <strong>github.event.action</strong>" of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> to fetch the activity type of the <strong>repository_dispatch</strong> event.</p>
<p>The activity type ( <strong>event_type</strong> , Required) is a custom webhook event name define in the <strong>request body</strong> when you run the " <strong>Create a repository dispatch event</strong>" API. You can set the name as any you like.</p>
<p>When using the API to trigger a workflow on the <strong>repository_dispatch</strong> event, the data you set in the <strong>request body</strong> will be passed into the <strong>github context</strong> of the triggered workflow run. The value of " <strong>event_type</strong>" will be passed to " <strong>github.event.action</strong>", and " <strong>client_payload</strong>" will be copied into the github context as a sub-object of " <strong>github.event</strong>".</p>
<p>Related docs:</p>
<p><a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#external-events-repository_dispatch" rel="nofollow noopener"><strong>External events: repository_dispatch</strong></a></p>
<p><a href="https://developer.github.com/v3/repos/#create-a-repository-dispatch-event" rel="nofollow noopener"><strong>Create a repository dispatch event</strong></a></p>
